packages/governance/test/swingsetTests/contractGovernor/governedContract.js

```diff
@@ -0,0 +1,56 @@
+// @ts-check
+
+import { Far } from '@agoric/marshal';
+import { E } from '@agoric/eventual-send';
+import { assert, details as X } from '@agoric/assert';
+import { sameStructure } from '@agoric/same-structure';
+
+import { buildParamManager, ParamType } from '../../../src/paramManager';
+
+const MALLEABLE_NUMBER = 'MalleableNumber';
+
+export const governedParameterTerms = {
+  contractParams: [MALLEABLE_NUMBER],
+};
+
+/** @type {ContractStartFn} */
+const start = async zcf => {
+  const {
+    /** @type {Instance} */ electionManager,
```

Comment:
Attn @mhofman @michaelfig

Reply:
> Can you help me figure out how/why `yarn lint` isn't complaining

We don't have "Typescript noImplicitAny" enabled.

> and VS Code isn't seeing it?

We don't have the JSDoc linting rules cranked up high enough to diagnose this questionable syntax.  Since this syntax of typing the destructuring isn't supported, VSCode doesn't complain (which is a major feature of using JSDoc).

> The syntax looks right to me

It doesn't look right to me, but that's probably because I'm conditioned by using VSCode, and this attempt to assign types to destructured properties has never worked.

I think the portable syntax you're reaching for is:

```js
  /** @type {{ electionManager: Instance, governedParams: ParameterNameList }} */
  const { electionManager, governedParams } = zcf.getTerms();
```

Even better would be to have a regularly-defined `@typedef` to name the terms for this contract, and use that defined name directly in `/** @type {MyKindOfTerms} */`.

> and WebStorm seems happy.

I don't know why WebStorm is happy.  It must not be using the Typescript Language Server which so many other systems rely on.  Or maybe your IDE is somehow using a different version of Typescript?

Reply:
I couldn't get `/** @type {{ electionManager: Instance, governedParams: ParameterNameList }} */` to work (standalone or as a typedef), and I couldn't find any contracts that declared the types of the values returned from `zcf.getTerms()`, so I dropped the declaration with disappointment.